src/pages/index.js

```diff
@@ -4,81 +4,191 @@ import PopupWithForm from "../components/PopupWithForm.js";
       const cardElement = createCard(data);
-      cardSection.addItem(cardElement);
+      if (cardElement) cardSection.addItem(cardElement);
     },
   },
   ".cards__list"
 );
-cardSection.renderItems();
 
-// Popup for image preview
+// ✅ Fetch and display only the current user's cards
+api
+  .getInitialCards()
+  .then((cards) => {
+    if (!Array.isArray(cards)) {
+      throw new Error("Invalid API response: Expected an array of cards");
+    }
+    cardSection.renderItems(cards);
+  })
+  .catch((err) => console.error("Error loading initial cards:", err));
+
+// ✅ Popup for image preview
 const imagePopup = new PopupWithImage(".modal_type_image");
 imagePopup.setEventListeners();
 
-// Profile popup
-const userInfo = new UserInfo({
-  nameSelector: ".profile__title",
-  jobSelector: ".profile__description",
-});
-
+// ✅ Profile popup
 const profilePopup = new PopupWithForm(
   ".modal_type_edit-profile",
   (data) => {
-    userInfo.setUserInfo({
-      name: data.name,
-      job: data.description,
-    });
+    return api
+      .updateUserInfo({
+        name: data.name,
+        about: data.description,
+      })
+      .then((updatedUser) => {
+        userInfo.setUserInfo({
+          name: updatedUser.name,
+          job: updatedUser.about,
+        });
+        document.forms["profile-form"].reset();
+      })
+      .catch((err) => console.error("Error updating user info:", err));
   },
   formValidators["profile-form"]
 );
 profilePopup.setEventListeners();
 
-// Add card popup
+// ✅ Add card popup (Only adds new card, does not re-fetch all)
 const addCardPopup = new PopupWithForm(
   ".modal_type_add-card",
   (data) => {
-    const newCard = createCard({ name: data.placeName, link: data.imageLink });
-    cardSection.addItem(newCard); // Add the new card to the DOM
-    if (formValidators["add-card"]) {
-      formValidators["add-card"].disableButton(); // Disable the button after submission
-    } else {
-      console.error('Validator for "add-card" is not defined');
-    }
+    return api
+      .addCard({ name: data.placeName, link: data.imageLink })
+      .then((newCard) => {
+        if (!newCard || !newCard._id) {
+          throw new Error("Error: New card does not have a valid ID.");
+        }
+
+        const cardElement = createCard(newCard); // ✅ Create only the new card
+        if (cardElement) cardSection.addItem(cardElement); // ✅ Add only the new card
+
+        document.forms["add-card"].reset();
+        formValidators["add-card"].disableButton();
+      })
+      .catch((err) => console.error("Error adding card:", err));
   },
   formValidators["add-card"]
 );
 addCardPopup.setEventListeners();
 
-// Event listeners for popups
+// ✅ Avatar update popup
+const avatarPopup = new PopupWithForm(
+  ".modal_type_avatar",
+  (data) => {
+    return api
+      .updateUserAvatar({ avatar: data.avatar })
+      .then((res) => {
+        document.querySelector(".profile__avatar").src = res.avatar;
```

Comment:
Only methods of `UserInfo` should set/get the profile data (including the **avatar**) 


You should use `userInfo.setUserInfo` here for setting the avatar